pkg/ansible/proxy/proxy.go

```diff
@@ -72,6 +72,12 @@ func CacheResponseHandler(h http.Handler, informerCache cache.Cache, restMapper
 				break
 			}
 
+			// check if resource doesn't exist in watched namespaces
+			// if watchedNamespaces[0] == "" then we are watching all namespaces
+			if watchedNamespaces[0] != "" && !contains(watchedNamespaces, r.Namespace) {
```

Comment:
It's not unreasonable that there could be a long list of namespaces, well into the thousands or beyond. Scanning such as how `contains` works could end up being expensive. It would be better if this could use a map so that membership is determined with an index.

Reply:
Great point. Easy change!